import "hero_chip.proto";
//武将升级，使用道具 command:103
message HeroUpgradeWithItemRequest
{
    required int32 hero_no = 1;
    required int32 exp_item_no = 2;
    required int32 exp_item_num = 3;
}

//武将突破 command:104
message HeroBreakRequest
{
    required int32 hero_no = 1;
}

//武将和碎片献祭 command:105
message HeroSacrificeRequest
{
    repeated int32 hero_nos = 1;
    repeated HeroChipPB hero_chips = 2;
}

//武将合成 command:106
message HeroComposeRequest
{
    required int32 hero_chip_no = 1;
}

//武将出售 command:107
message HeroSellRequest
{
    repeated int32 hero_nos = 1;
}

//hero refine
message HeroRefineRequest
{
    required int32 hero_no= 1;
    required int32 refine = 2;
}
// 武将一键升级请求, command:120
message HeroRequest
{
    required int32 hero_no = 1;
}

// 武将觉醒请求, command:121
message HeroAwakeRequest
{
    required int32 hero_no = 1; // 武将编号
    optional int32 awake_item_num = 2; // 觉醒道具数量
}
// 武将突破信息同步, command 122
message HeroBreakRelatedInfoRequest
{
    required int32 hero_no = 1; // 武将编号
}
